tests/ludwig/modules/test_convolutional_modules.py

```diff
@@ -1,11 +1,109 @@
 import pytest
 from typing import Callable, Dict, List, Optional, Tuple, Union
-
+import numpy as np
 import torch
 
-from ludwig.modules.convolutional_modules import Conv2DLayer, Conv2DStack,\
-    Conv2DLayerFixedPadding, ResNetBlock, ResNetBottleneckBlock,\
-    ResNetBlockLayer, ResNet
+from ludwig.modules.convolutional_modules import Conv2DLayer, Conv2DStack, \
+    Conv2DLayerFixedPadding, ResNetBlock, ResNetBottleneckBlock, \
+    ResNetBlockLayer, ResNet, Conv1DLayer, Conv1DStack, ParallelConv1DStack, \
+    ParallelConv1D
+
+BATCH_SIZE = 2
+SEQ_SIZE = 100
+HIDDEN_SIZE = 64
+
+
+###
+# Helper function to compute expected output shape
+# for Conv1D related layers
+###
+def expected_seq_size(
```

Comment:
You could use the [get_img_output_shape](https://github.com/ludwig-ai/ludwig/blob/81f86c1623646064cbabde716ee45b5f29f98663/ludwig/utils/image_utils.py#L179) function for this within image_utils.py. That function operates on convs and pools independently, so you could call the function twice if you wanted to perform both a conv followed by a pool.

Reply:
As I understand the implementation of `get_img_output_shape`, it is geared for a 2D structure.  To adapt it for the 1D structures, do I set, for example, `img_height` to zero?

Reply:
On a lark, I tried setting `img_height` to zero and it works.  Here is the code structure:
```
    output_seq_size = get_img_output_shape(
        0,  # img_height set to zero for 1D structure
        SEQ_SIZE,  # img_width equates to sequence size
        kernel_size, # kernel_size for conv layer
        strides,  # stride for conv layer
        padding,  # padding for conv layer
        dilation  # dilation for conv layer
    )
    if pool_size is not None:
        # adjust expected output shape for pooling layer
        output_seq_size = get_img_output_shape(
            0,  # img_height set to zero for 1D structure
            output_seq_size[1],  # img_width equates to sequence size
            pool_size,  # kernel size for pooling layer
            pool_stride,  # stride for pooling layer
            pool_padding,  # padding for polling layer
            1  # pooling layer only support unit dilation
        )
    assert out_tensor.size() == (BATCH_SIZE, output_seq_size[1], NUM_FILTERS)

```

if this is what you had in mind, I can retire the `expected_seq_size()` or actually change the implementation in that function to use the `get_img_output_shape()` function instead of the hand coded formulas.

Reply:
Nice catch with setting `img_height=0`!

Yep, that's what I had in mind. Up to you -- I would be ok with either choice. Could you also remove the comments in the arguments, and use keyword args instead?

Reply:
done.